.github/workflows/validator-tests.yaml

```diff
@@ -0,0 +1,41 @@
+name: Validator Tests
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  check-schema:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+
+    - name: Set up Go
+      uses: actions/setup-go@v5
+      with:
+        go-version: '~1.22'
+
+    # TODO: Publish releases
+    - name: Build
+      run: make validator
+
+    - name: Test
+      run: |
+        cd validator
+        go test .
+
+    - name: Run ShellTests
+      run: |
+        cd validator
+
+        sudo apt-get update
+        sudo apt-get install shelltestrunner
+
+        shelltest -c --diff --all shelltests/*.test
```

Comment:
Is it feasible to include this in the makefile? Can `shelltest` be used on macos?

Reply:
It looks like `Dockerfile.shelltest` is aimed at running these tests inside a docker container. If that's what we expect users to run locally, we should probably run the same in the build for consistency.